### ISSUE TITLE
urlKeys not working for logging in

### ISSUE THREAD

**Issue by johnfmorton** [sjelfull/craft3-autologin#12]
### Describe the bug

I can't successfully log in using the `urlKeys` feature on my remote servers, but when I'm in my local development environment, it works.

### Steps to reproduce

An example from my `autologin.php` file:

```
// A list of Craft usernames/emails mapped to URL keys
'urlKeys' => [
  'admin' => 'my-secret-key',
],
```

With this configuration on my production server, I would expect to be able to go to 

```
https://siteurl.tld/autologin?key=my-secret-key
```

and be logged in as the admin user. Instead, I am presented with a blank screen, even when `devMode` is set to `true` on the production server.

The local version works fine, though, with a URL like this:

```
localhost/autologin?key=my-secret-key
```

Back on the public server, if I try to add the `cp=true` parameter to the URL, like this

```
https://siteurl.tld/autologin?key=my-secret-key&cp=true
```

I am redirected to the control panel with the username and password field not filled in. 

As before, the localhost version in my dev environment works as expected.

```
https://localhost/autologin?key=my-secret-key&cp=true
```

## ipWhitelist does work in production

Interestingly, the `ipWhitelist` feature does work in production. 

```
'ipWhitelist' => [
  'admin' => [
    '123.456.123.45',
  ],
],
```

Having this in the autologin.php always works successfully on the production server. 

### Craft CMS version

4.3.9

### Plugin version

3.0.0

### Multi-site?

No

### Additional context

As I mentioned above, this works in my local development environment but fails in production.

**Comment by engram-design** [comment]
Hmm, difficult to say. I'll need to add some logging to the entire process (there's currently none) to see what's going on.

**Comment by johnfmorton** [comment]
I have figured out why the autologin was not working for my production server. 

### **TLDR:** FastCGI Cache with an NGINX config not configured for the 'autologin' route.

Here's the longer version in case anyone else stumbles upon this page.

I've got my site configured with FastCGI Cache as described here: https://nystudio107.com/blog/static-caching-with-craft-cms

I added two rules to my NGINX.config to allow the Autologin plugin to work:

## First update

I added `autologin` to the URLs that would set the `$no_cache` variable to true.

Before the edit, this rule looked like this:

```
if ($request_uri ~* "/(admin/|cpresources/)")
{
    set $no_cache 1;
}
```

After the update, it looked like this:

```
if ($request_uri ~* "/(admin/|cpresources/|autologin)")
{
    set $no_cache 1;
}
```

This rule alone doesn't fix the issue but ensures these requests don't end up in your FastCGI Cache.


## Second update

Next, I needed to make the Autologin attempts routed through the `index.php` file. I added this rule to the config file:

```
 # Ensure the autologin plugin functionality requests route through index.php
location ^~ /autologin {
    try_files $uri $uri/ @phpfpm_nocache;
}
```

After applying both rules to my site, the autologin feature works successfully.
